phoenix-core/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java

```diff
@@ -610,7 +610,9 @@ public static void setTimestamps(MiniBatchOperationInProgress<Mutation> miniBatc
             }
             Mutation m = miniBatchOp.getOperation(i);
             // skip this mutation if we aren't enabling indexing or not an atomic op
-            if (!builder.isEnabled(m) && !builder.isAtomicOp(m)) {
+            // or if it is an atomic op and its timestamp is already set
+            if (!builder.isEnabled(m) &&
+                (!builder.isAtomicOp(m) || getMaxTimestamp(m) != HConstants.LATEST_TIMESTAMP)) {
```

Comment:
Should this be && rather than ||? The comment says "and" not "or"

Reply:
Agree with gokcen, something looks missing here.

Reply:
@ankitjain64 @gokceni It is correct but I will restructure it to make it more readable